code/modules/mob/living/carbon/human/human.dm

```diff
@@ -150,15 +147,19 @@
 				Paralyse(10)
 
 	take_overall_damage(b_loss,f_loss)
-
+	//attempt to dismember bodyparts
+	for(var/X in bodyparts)
+		var/obj/item/bodypart/BP = X
+		if(prob(50/severity) && BP.body_zone != "head" && BP.body_zone != "chest")
+			BP.dismember()

```

Comment:
May want to limit the amount of parts that can dismember based on explosion severity. You really shouldn't lose all your limbs form a light, 30 damage explosion just because of probability check. LIkewise, bomb armor should probably reduce or eliminate the chances of this happening.